File: 灵茶/滑动窗口与双指针/定长窗口/2090.py
class Solution:
    def getAverages(self, nums: List[int], k: int) -> List[int]:
        if k ==0:
            return nums
        sum = 0
        ans = [0] * len(nums)
        
        for index, num in enumerate(nums):
            sum += num
            if index >= 2 * k:
                ans[index - k] = int (sum / (2 * k + 1))
                sum -= nums[index - 2 * k]
            ans[index] = -1


        return ans



        

File: 灵茶/滑动窗口与双指针/定长窗口/2090.java
class Solution {
    public int[] getAverages(int[] nums, int k) {
        if (k == 0) {
            return nums;
        }
        Long sum = 0L;
        int[] result = new int[nums.length];
        for (int i = 0; i < nums.length; i++) {
            sum += nums[i];
            if (i >= 2 * k) {
                result[i - k] = (int) (sum / (2 * k + 1));
                sum -= nums[i - 2 * k];
            }
            result[i] = -1;
        }
        return result;
    }
}